model/task_lifecycle.go

```diff
@@ -604,10 +561,20 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		finishedTasks++
 		tasksToNotify += 1
 
+		now = time.Now()
 		displayTask, err = t.GetDisplayTask()
 		if err != nil {
 			return errors.WithStack(err)
 		}
+		if time.Since(now) > slowMS {
```

Comment:
For the ones in this loop you might want to log the task you're on as well

Reply:
done